skore/src/skore/sklearn/_estimator/metrics_accessor.py

```diff
@@ -448,6 +455,238 @@ def _compute_metric_scores(
+        ...     LogisticRegression(),
+        ...     X_train=X_train,
+        ...     y_train=y_train,
+        ...     X_test=X_test,
+        ...     y_test=y_test
+        ... )
+        >>> report.metrics._fit_time()
+        {'fit': ...}
+
+        >>> estimator = LogisticRegression().fit(X_train, y_train)
+        >>> report = EstimatorReport(estimator, X_test=X_test, y_test=y_test)
+        >>> report.metrics._fit_time()
+        Traceback (most recent call last):
+        skore.sklearn._estimator.metrics_accessor.FitInformationUnavailableError: ...
+        """
+        fit_time = self._parent._cache.get(("fit_time",))
+        if fit_time is None:
+            raise FitInformationUnavailableError(
+                "The fit information is not available, perhaps because the estimator "
+                "was fitted before the report was created."
+            )
+        return {"fit": fit_time}
+
+    def _predict_time(
+        self,
+        *,
+        data_source: DataSource = "test",
+        X: Optional[ArrayLike] = None,
+        y: Optional[ArrayLike] = None,
+    ) -> dict:
+        """Get the time taken to compute the estimator's `predict` method.
+
+        Will compute the predictions if necessary.
+
+        Examples
+        --------
+        >>> from sklearn.datasets import load_breast_cancer
+        >>> from sklearn.linear_model import LogisticRegression
+        >>> from sklearn.model_selection import train_test_split
+        >>> from skore import EstimatorReport
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...     *load_breast_cancer(return_X_y=True), random_state=0
+        ... )
+        >>> classifier = LogisticRegression(max_iter=10_000)
+        >>> report = EstimatorReport(
+        ...     classifier,
+        ...     X_train=X_train,
+        ...     y_train=y_train,
+        ...     X_test=X_test,
+        ...     y_test=y_test,
+        ... )
+
+        >>> report.metrics._predict_time()  # Predictions on the test data
+        {'predict_test': ...}
+
+        >>> report.metrics._predict_time(data_source="train")
+        {'predict_train': ...}
+
+        >>> report.metrics._predict_time(data_source="X_y", X=X_test, y=y_test)
+        {'predict_X_y_f41066b51130a331835e7d66417ff728': ...}
+        """
+        X, y, data_source_hash = self._get_X_y_and_data_source_hash(
+            data_source=data_source, X=X, y=y
+        )
+
+        def find_in_cache():
+            # Search for first matching key in cache
+            if data_source_hash is None:
+                for key in self._parent._cache:
+                    if key[-2:] == (data_source, "predict_time"):
+                        return self._parent._cache[key]
+            else:
+                for key in self._parent._cache:
+                    if key[-3:] == (data_source, data_source_hash, "predict_time"):
+                        return self._parent._cache[key]
+            return None
```

Comment:
Could you make it a private method of the class. By changing the code in the `_base.py`, I think that we can avoid the `if/else` and have a single branch`. Also, we could create they key instead to have to slice it:

```python
expected_key = (self._parent._estimator_hash, data_source, data_source_hash or None, "predict_time")
if expected_key in self._parent._cache:
    return ...
return None
```

I assume that we will get a performance gain when checking the keys of the `_cache` directly.